third_party/amd/lib/TritonAMDGPUToLLVM/TargetInfo.cpp

```diff
@@ -126,10 +126,146 @@ Value TargetInfo::programId(RewriterBase &rewriter, Location loc,
   return LLVM::AMD::llGetPid(loc, rewriter, moduleOp, axis);
 }
 
+static std::optional<Operation *> getReduxOpForDPP(triton::ReduceOp op) {
```

Comment:
I'm not sure the purpose of `std::optional`--it's quite common to return `nullptr` to mean not found for pointer returns.